sdk/lib/atl/atlpath.h

```diff
@@ -23,10 +23,10 @@ class CPathT
     // const
     inline BOOL PathFileExistsX(LPCSTR pszPath) { return PathFileExistsA(pszPath); }
     inline BOOL PathFileExistsX(LPCWSTR pszPath) { return PathFileExistsW(pszPath); }
-    inline int PathFindExtensionX(LPCSTR pszPath) { return PathFindExtensionA(pszPath); }
-    inline int PathFindExtensionX(LPCWSTR pszPath) { return PathFindExtensionW(pszPath); }
-    inline int PathFindFileNameX(LPCSTR pszPath) { return PathFindFileNameA(pszPath); }
-    inline int PathFindFileNameX(LPCWSTR pszPath) { return PathFindFileNameW(pszPath); }
+    inline PVOID PathFindExtensionX(LPCSTR pszPath) { return PathFindExtensionA(pszPath); }
```

Comment:
I'm surprised the old code worked with any compiler... but I'm also surprised that this works. The function returns a pointer-to-const, so converting it to PVOID would need a cast.
Any reason we're not just using the correct return type here?

Reply:
I guess nobody cared until now because that code is not instantaited/optimized away ?

Reply:
It should be perfectly legal here. You can't overload _only_ the return type. But as long as parameter types differ between overloads, you can change the return type, too.